python/ray/tests/test_advanced_3.py

```diff
@@ -124,6 +124,67 @@ def f(x):
     attempt_to_load_balance(f, [x], 100, num_nodes, 25)
 
 
+def test_locality_aware_leasing(ray_start_cluster):
```

Comment:
Nice test! Minor, but you can clean this up a bit with some of the existing utils.
- Use custom resources to pin `f` to one of the nodes. You can set custom resources on one of the worker nodes by using `cluster.add_node(resources={"custom": 1})`.
- You can use ray.test_utils.wait_for_condition for the timeout (but probably not needed if you do the above).

Reply:
Ah yes using custom resources to pin that first task to a specific node will be way, way nicer! And I didn't use `wait_for_condition` given that I needed the return value of the predicate, but no condition waiting should be needed when using a custom resource constraint to pin the first task. I'll push that up.